axum/src/routing/path_router.rs

```diff
@@ -36,6 +36,69 @@ where
     }
 }
 
+// Validates a path in compile time, used with the vpath macro.
+// Allow dead code needed because this is only used within the vpath macro
+// and if there's no call to it, then, there's no users of it.
+#[allow(dead_code)]
+#[doc(hidden)]
+pub const fn __private_validate_static_path(path: &'static str) -> &'static str {
+    if path.is_empty() {
+        panic!("Paths must start with a `/`. Use \"/\" for root routes")
+    }
+    if path.as_bytes()[0] != b'/' {
+        panic!("Paths must start with /");
+    }
+    path
+}
+
+#[rustversion::since(1.80)]
+#[macro_export]
+/// This macro checks that your string is the correct format to be used as a route,
+/// and fails in compilation time instead of runtime, if it's not.
```

Comment:
This is a slightly confusingly-written sentence. Would recommend perhaps:

> Checks that a path is usable as a route at compile time.

Reply:
I blame my brazilian upbringing :)
will update.